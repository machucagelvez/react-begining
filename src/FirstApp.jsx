import PropTypes from 'prop-types'

// Aun estando por fuera sigue siendo parte del componente
// Así no se tiene que re-renderizar cuando se hace un cambio dentro del componente
// Sí no hay una dependencia internamente se recomienda poner fuera del componente
// const newMessage = {message: 'Hola mundo', title: 'Ozzy'}

// const getMessage = () => { 
//   return 'Hola Mundo'
// }

export const FirstApp = ({title, subTitle, name}) => {
  // console.log(props)

  return (
    <>
      {/* Con funciones asíncronas se genera un error porque devuelve un objeto: */}
      <h1 data-testid="test-title"> {title} </h1>
      {/* <h1>{getMessage()}</h1> */}
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

// Los PropTypes sirven para tipar las props
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  name: 'Ozzy',
  subTitle: 'No hay subtítulo',
  // title: 'No hay título',
}
