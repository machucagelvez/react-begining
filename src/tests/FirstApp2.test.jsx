import { render, screen } from "@testing-library/react"
import { FirstApp } from "../FirstApp"

describe('Test in FirstApp', () => { 
  const title = 'Buenas'
  const subTitle = 'Esto es un subtÃ­tulo'

  test('should match with snapshot', () => { 
    const { container } = render(<FirstApp title={title} />)
    expect(container).toMatchSnapshot()
  })

   test('should show message "Buenas"', () => { 
    render(<FirstApp title={title} />)
    expect(screen.getByText(title)).toBeTruthy()
    })

   test('should show title in h1', () => { 
    render(<FirstApp title={title} />)
    expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)
    })

   test('should show subtitle sended by props', () => { 
    render(<FirstApp title={title} subTitle={subTitle} />)
    expect(screen.getAllByText(subTitle).length).toBe(2)
    })
})