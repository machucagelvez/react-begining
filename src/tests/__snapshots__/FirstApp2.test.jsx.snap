// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in FirstApp should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Buenas
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Ozzy
  </p>
</div>
`;
